docs/DeveloperGuide.adoc

```diff
@@ -46,177 +39,186 @@ The following class plays an important role at the architecture level:
-.  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
-
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
-
-.Interactions Inside the Logic Component for the `delete 1` Command
-image::DeleteSequenceDiagram.png[]
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
-image::ModelClassDiagram.png[]
+image::ModelClassDiagram_FlashCardPro.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
+*API* : link:{repoURL}/src/main/java/dream/fcard/model[`Model`]
 
-The `Model`,
+.Structure of the Model Component
+image::cards_class_diagram.png[]
 
-* stores a `UserPref` object that represents the user's preferences.
-* stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
-* does not depend on any of the other three components.
 
-[NOTE]
-As a more OOP model, we can store a `Tag` list in `Address Book`, which `Person` can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object. An example of how such a model may look like is given below. +
- +
-image:BetterModelClassDiagram.png[]
+.`FlashCard`,
+
+* There are 3 types of cards used by our App, the `JavascriptCard`, `FrontBackCard` and `MultipleChoiceCard`.
+* `JavascriptCard` and `FrontBackCard` implements from `FlashCard`, an interface class.
+* `MultipleChoiceCard` inherits from `FrontBackCard`.
+* The interface class `FlashCard` also implements `JsonInterface` class to allow loading and saving to Json format.
+
+.FlashCard Priority
+Each FlashCard has an associated priority level, which the user can indicate his/her value of the importance of the card.
+
+The priority levels are described below:
+
+[width="59%",cols="22%,<23%,<25%",options="header",]
+|=======================================================================
+| Priority| Value| Intended Action
+
+| HIGH_PRIORITY| 10| Valued as important by user, should be tested more often
+
+| LOW_PRIORITY| 1| Valued, but less important by user, should be tested less often
+
+|=======================================================================
+
+The sequence diagram below shows how the `ExamRunner` component can use the `createSubsetForTest()` method in the `Deck` and make use of the priorities associated with the `FlashCards` to create a test that emphasises the `FlashCards` that the user deems as important.
+
+image::flashcards_priority cards.png[]
```

Comment:
![image](https://user-images.githubusercontent.com/8232172/67543147-93ac4480-f722-11e9-9af9-9e1c954ba385.png)

I've noticed a few possible issues with this diagram:

- Can you enclose the "else" in brackets? I think that is proper UML notation.
- Would it be more proper to use two arrows to represent calling a self method, instead of using one?